src/SplitPane.js

```diff
@@ -1,361 +1,425 @@
+
+    this.setState({
+      sizes
+    });
   }
 
-  onMouseUp() {
-    const { allowResize, onDragFinished } = this.props;
-    const { active, draggedSize } = this.state;
-    if (allowResize && active) {
-      if (typeof onDragFinished === 'function') {
-        onDragFinished(draggedSize);
-      }
-      this.setState({ active: false });
+  setPaneRef = (idx, el) => {
+    if (!this.paneElements) {
+      this.paneElements = [];
     }
+
+    this.paneElements[idx] = el;
   }
 
-  setSize(props, state) {
-    const { primary } = this.props;
-    const ref = primary === 'first' ? this.pane1 : this.pane2;
-    let newSize;
-    if (ref) {
-      newSize =
-        props.size ||
-        (state && state.draggedSize) ||
-        props.defaultSize ||
-        props.minSize;
-      ref.setState({
-        size: newSize,
-      });
-      if (props.size !== state.draggedSize) {
-        this.setState({
-          draggedSize: newSize,
-        });
-      }
-    }
+  getResizersSize(children) {
+    return (children.length - 1) * this.props.resizerSize;
   }
 
   render() {
-    const {
-      allowResize,
-      children,
-      className,
-      defaultSize,
-      minSize,
-      onResizerClick,
-      onResizerDoubleClick,
-      paneClassName,
-      pane1ClassName,
-      pane2ClassName,
-      paneStyle,
-      pane1Style: pane1StyleProps,
-      pane2Style: pane2StyleProps,
-      primary,
-      prefixer,
-      resizerClassName,
-      resizerStyle,
-      size,
-      split,
-      style: styleProps,
-    } = this.props;
-    const disabledClass = allowResize ? '' : 'disabled';
-    const resizerClassNamesIncludingDefault = resizerClassName
-      ? `${resizerClassName} ${RESIZER_DEFAULT_CLASSNAME}`
-      : resizerClassName;
-
-    const style = Object.assign(
-      {},
-      {
-        display: 'flex',
-        flex: 1,
-        height: '100%',
-        position: 'absolute',
-        outline: 'none',
-        overflow: 'hidden',
-        MozUserSelect: 'text',
-        WebkitUserSelect: 'text',
-        msUserSelect: 'text',
-        userSelect: 'text',
-      },
-      styleProps || {}
-    );
+    const { children, className, split } = this.props;
+    const notNullChildren = removeNullChildren(this.props.children)
+    const sizes = this.getSizes();
+    const resizersSize = this.getResizersSize(notNullChildren);
 
-    if (split === 'vertical') {
-      Object.assign(style, {
-        flexDirection: 'row',
-        left: 0,
-        right: 0,
-      });
-    } else {
-      Object.assign(style, {
-        bottom: 0,
-        flexDirection: 'column',
-        minHeight: '100%',
-        top: 0,
-        width: '100%',
-      });
-    }
+    const elements = notNullChildren.reduce((acc, child, idx) => {
+      let pane;
+      const resizerIndex = idx - 1;
+      const isPane = child.type === Pane;
+      const paneProps = {
+        index: idx,
+        'data-type': 'Pane',
+        split: split,
+        key: `Pane-${idx}`,
```

Comment:
To me, an interesting improvement would be to let the user set the `key` for `Pane` components.
I implemented an add / remove logic for panes using a DOM structure similar to
```
<SplitPane>
  {panes.map((paneId: string) => 
    <Pane key={`scene-pane-${paneId}`}>
        ...
    </Pane>
  }
</SplitPane>
```
Here, `panes` is a state variable consisting of a list of unique pane identifiers. When I close a pane, a callback simply removes its id from the list. However, because this librairy rewrites `key` for `Pane`, I get inconsistent results.
Please consider using something like [`nanoid`](https://github.com/ai/nanoid/) to generate unique ids for panes rather than using their position in the list of panes.